### PR TITLE
ARC: ARCv3: add SMP support

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
- add support of SMP to ARCv3 64 bit port
- add nsim_hs6x_smp simulation board (nSIM based) with 2 cores ARCv3 HS6x CPU

### CODE REVIEW

**Review by abrodkin — APPROVED** [review]
With just one note about CPU name in `.dts` file LGTM.

File: boards/arc/nsim/nsim_hs6x_smp.dts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	model = "snps,nsim_hs";
+	compatible = "snps,nsim_hs";
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "snps,archs";
```
[the remark this answers is not in the document]
Comment (ruuddw) [reply boards/arc/nsim/nsim_hs6x_smp.dts:21]: That would hold for the single core as well, that already uses archs as well.
I don't have a strong opinion either way. The 64-bit hs6x is still an ARC HS family member.
And, we don't use bindings for this as far as I know, so the name is not really used anywhere.

**Review by ruuddw — APPROVED** [review]

File: boards/arc/nsim/nsim_hs6x_smp.dts (hunk 1 above)
[the remark this answers is not in the document]
Comment (evgeniy-paltsev) [reply boards/arc/nsim/nsim_hs6x_smp.dts]: @abrodkin changed to `snps,arc-hs6x`

[the remark this answers is not in the document]
Comment (ruuddw) [reply boards/arc/nsim/nsim_hs6x_smp.dts]: Can you update boards/arc/nsim/nsim_hs6x.dts to arc-hs6x as well then, please? (and, if introducing the detailed cpu variant here, shouldn't it be done for hs4x, hs3x, em4/5D/6/7D/9D/11D, as well?

[the remark this answers is not in the document]
Comment (evgeniy-paltsev) [reply boards/arc/nsim/nsim_hs6x_smp.dts]: > and, if introducing the detailed cpu variant here, shouldn't it be done for hs4x, hs3x, em4/5D/6/7D/9D/11D, as well?

Hmm, I've thought about it - probably it's not worth it as we don't use / check these compatible strings anywhere but with such change we'll need to maintain and set them properly for all the boards...

I've switched this board compatible to more generic `"snps,arcv3-hs"` for now. We can discuss the options on our weekly call.

PS: The changes for `boards/arc/nsim/nsim_hs6x.dts` need to be done in a separate PR.

**Pull request merged**